### ISSUE TITLE
Conda: ImportError when importing pyvinecopulib in Python 3.7

### ISSUE THREAD

**Issue by dmey** [vinecopulib/pyvinecopulib#58]
The following error is raised when trying to import pyvinecopulib in Python 3.7 on anaconda:

```
>>> import pyvinecopulib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
```

### How to reproduce
Create file named `environment.yml` with the following content:

``` yml
name: pyvinecopulib-py37
channels:
  - defaults
  - conda-forge
dependencies:
  - python=3.7
  - pip
  - setuptools
  - pybind11
  - pip:
    - pyvinecopulib
```

Then create the environment with `conda env create -f environment.yml` and activate with `conda activate pyvinecopulib-py37` from your Anaconda Prompt. Then launch `python` and `import pyvinecopulib`.

**Comment by tnagler** [comment]
``` shell
sudo apt-get install libpython3.7
```
did the trick on linux

**Comment by tnagler** [comment]
``` shell
export LD_LIBRARY_PATH=~/anaconda3/lib
```
also seems to work (no idea why conda isn't doing that automatically).

**Comment by dmey** [comment]
> ```shell
> sudo apt-get install libpython3.7
> ```
> 
> did the trick on linux

yes but now you are linking and relying on the system's Python. On Linux and macOS it is advisable not to link against `libpython` when building you package as this may create conflicts when more than one version of Python are installed (see note in https://pybind11.readthedocs.io/en/stable/compiling.html#building-manually).

**Comment by tnagler** [comment]
We're not linking against anything ourselves, we only include headers in our `setup.py`. But [pybind11 itself is linking against `libpython`](https://pybind11.readthedocs.io/en/stable/faq.html#cmake-doesn-t-detect-the-right-python-version) and it has to do that to create a valid python library (as far as I understand). 

While pybind11 figures out where `libpython` lives, conda does not add its `libpython` to the search path, which seems to create the problem.

**Comment by dmey** [comment]
I am not familiar with `pybind11` but from what I understand linking against `libpython` should be avoided for the reason I described above. @letmaik, you have much more experience than me with this sort of stuff -- have you seen this issue before? Maybe you can shed some light on the problem might be...

**Comment by letmaik** [comment]
I haven't built pybind extension modules directly from setup.py but rather outside using CMake, which likely explains the difference. pybind's CMake does the right thing, and Python historically (up until 3.8) did the "wrong" thing (or, less compatible thing). Python changed this in 3.8: "On Unix, C extensions are no longer linked to libpython except on Android and Cygwin." (https://docs.python.org/3/whatsnew/3.8.html#debug-build-uses-the-same-abi-as-release-build).

Long story short, this issue will go away automatically with Python 3.8 and there's nothing that can be done to prevent linking against libpython with <3.8 since Python is in control of the extension build process when using setup.py.

My suggestion is to release Python 3.8 wheels to PyPI and switch to 3.8 when using this library with conda.

**Comment by tvatter** [comment]
Python 3.8 wheels are on PyPI with #61, I think that it closes this.
